frontend/src/metabase/querying/notebook/components/JoinStep/JoinConditionDraft/JoinConditionDraft.tsx

```diff
@@ -77,6 +77,13 @@ export function JoinConditionDraft({
     handleColumnChange(lhsColumn, newRhsColumn);
   };
 
+  useLayoutEffect(() => {
+    setLhsColumn(undefined);
+    setRhsColumn(undefined);
+    setIsLhsOpened(true);
+    setIsRhsOpened(false);
+  }, [joinable]);
```

Comment:
Does this rely on the referential integrity of the `joinable`? 
And if so, is that safe?

Reply:
That's a good point. I modified it in a way so we rely on the identify of `rhsTable` only which we store in react state for draft joins.